### ISSUE TITLE
AsyncRecord in AsyncProxyStrategy has incomplete API

### ISSUE THREAD

**Issue by jakubkosinski** [RailsEventStore/rails_event_store#132]
In order to make `AsyncRecord` work with AR, it should respond to more methods:
- `rolledback!`
- `before_committed!`

They can just return `nil` but should be defined.

* [x] `v0.18.2`
* [x] `master`

**Comment by paneq** [comment]
@jakubkosinski which rails version are we talking about?

**Comment by paneq** [comment]
I was able to reproduce on Rails 5.1.4

```
(fix-missing-methods) > RAILS_VERSION=5.1.4 bundle exec rspec
5
1
...before_committed!
committed!
.rolledback!
........................
```

the reason it was failing for you is that we were not testing on this rails version so far. The tests in general were going through this path in code.
